### PR TITLE
refactor: improve modularity

### PR DESCRIPTION
**Pull request by tianzhou**
* introduce cobra to handle flags
* signal handling

The idea is to make flag namespaced based on module

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,16 +1,19 @@
 module github.com/bytebase/relay
 
-go 1.19
+go 1.17
```
Comment (unknwon) [inline go.mod:3]: Do we need to be compatible with 1.17?

**Review by unknwon — APPROVED** [review]
Few minor issues, approve to unblock

File: cmd/root.go [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+	"net/http"
+	"os"
+	"os/signal"
+	"strings"
+	"syscall"
+
+	"github.com/bytebase/relay/util"
+	"github.com/flamego/flamego"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
```
[the remark this answers is not in the document]
Comment (tianzhou) [reply cmd/root.go:20]: It's using go's tool

File: main.go [hunk 3]
```diff
@@ -1,107 +1,13 @@
...
-	defer func() { _ = resp.Body.Close() }()
-
-	if resp.StatusCode/100 != 2 {
-		body, err = io.ReadAll(resp.Body)
-		if err != nil {
-			return errors.Wrap(err, "read response body")
-		}
-		return errors.Errorf("unexpected status code %d with body: %s", resp.StatusCode, body)
+	if err := cmd.Execute(); err != nil {
+		os.Exit(1)
```
[the remark this answers is not in the document]
Comment (tianzhou) [reply main.go:13]: Done

**Review by tianzhou — COMMENTED** [review]
PTAL

**Pull request merged**